### ISSUE TITLE
There is a remote code execution (RCE) vulnerability exists in ujcms v6.0.2

### ISSUE THREAD

**Issue by Sma11New** [ujcms/ujcms#5]
[Vulnerability description]
The Remote Code Execution (RCE) vulnerability exists in ujcms v6.0.2, when the project is partially configured on Linux using Tomcat, attackers can use path traversal and arbitrary file uploads to execute arbitrary code.

[Vulnerability Type]
Remote Code Execution (RCE)

[Vendor of Product]
https://gitee.com/ujcms/ujcms
https://github.com/ujcms/ujcms
https://www.ujcms.com/

[Affected Product Code Base]
v6.0.2

[Vulnerability proof]
The code restricts the access and execution of jsp and jspx files, but it exists in any path and any file upload, so you can upload a web.xml file and add a jsp resolvable suffix, such as abc
```
<servlet-mapping>
    <servlet-name>jsp</servlet-name>
    <url-pattern>*.abc</url-pattern>
</servlet-mapping>
```
Condition: It needs to be deployed with tomcat on Linux, and the configuration file cannot be overwritten on Windows, and File.renameTo is used

1. Upload web.xml
Download an initial configuration file web.xml of tomcat, and add the above configuration in the following location
![image](https://github.com/ujcms/ujcms/assets/53944964/ec556493-005f-4839-89a2-85b62372437d)
Upload the web.xml file to the uploads directory first, and use the path id to indicate it when uploading, and the path cannot be traversed
![image](https://github.com/ujcms/ujcms/assets/53944964/085b818e-52c0-4c2b-ae49-a8a0ccadc963)
Rename, the capture package can traverse the path at the file name, and overwrite the original web.xml
![image](https://github.com/ujcms/ujcms/assets/53944964/29e0c230-e1af-4dbd-a4eb-49da226f7992)

2. Upload the Trojan horse
Upload a Trojan horse and execute the ping command. At this time, the uploaded suffix cannot be the custom analytical suffix above. You can upload any suffix and rename it later, otherwise the upload will not succeed
![image](https://github.com/ujcms/ujcms/assets/53944964/c46486f0-f50d-4a22-ac0f-9ac71591b510)
The upload suffix is ​​abc1
![image](https://github.com/ujcms/ujcms/assets/53944964/f1c37930-2fb6-4698-8d97-df89a49cec4f)
Renamed to ../../123.abc, the path traverses to the root directory
![image](https://github.com/ujcms/ujcms/assets/53944964/55fc6220-dce6-4d43-bf50-24fd426e5c1a)
Visit 123.abc and successfully trigger rce
![image](https://github.com/ujcms/ujcms/assets/53944964/75e0db2d-1247-4025-ba10-2cf602814f04)

[Code Details]
1. Upload
Track the rename interface, find that the parameters are being passed to doUpload, and verify the suffix at the upload
com.ujcms.cms.core.web.backendapi.AbstractUploadController#doUpload
![image](https://github.com/ujcms/ujcms/assets/53944964/acdbeabe-4e7a-409d-847f-a6cf0f9ed735)
The suffix can be uploaded without any problem.

2. Rename
com.ujcms.cms.core.web.backendapi.AbstractWebFileController#rename is the same as upload, it has checkName to verify the file name, enter here to view the code
![image](https://github.com/ujcms/ujcms/assets/53944964/947820b0-3a36-4892-aa14-3fddb171fd9a)
com.ujcms.cms.core.web.backendapi.AbstractWebFileController#checkName(java.lang.String) Check the file name, when both meet
    (1) The file name is empty
    (2) The file name contains illegal characters
an exception is thrown, but the first condition is always false, and the conditions cannot be met at the same time, so this verification will always be bypassed, so it can be executed, and then any file can be renamed and uploaded.
![image](https://github.com/ujcms/ujcms/assets/53944964/535ab250-6f94-463b-963e-6caad381ee4a)
So there is a problem with the judgment logic here, it should be || instead of &&, which leads to the failure of the security check in this place. Of course, this is also the reason why it can be used successfully.
After passing the verification of this block, use File.renameTo in com.ujcms.util.file.LocalFileHandler#rename to rename the file.
![image](https://github.com/ujcms/ujcms/assets/53944964/693ca05b-73c8-487e-b892-7093866ea6f7)
renameTo cannot overwrite files in Windows, but can overwrite and create directories in Linux, so this vulnerability can only be exploited in Linux.

**Comment by ujcms** [comment]
Thank you, this problem will be fixed in the next version

**Comment by ujcms** [comment]
Fixed in version 7.0.0
